kadi/commands/commands_v2.py

```diff
@@ -1315,26 +1315,12 @@ def get_matching_block_idx(cmds_arch, cmds_recent):
     idx_arch_recent = cmds_arch.find_date(cmds_recent["date"][0])
     logger.info("Selecting commands from cmds_arch[{}:]".format(idx_arch_recent))
     cmds_arch_recent = cmds_arch[idx_arch_recent:]
+    cmds_arch_recent.rev_pars_dict = weakref.ref(REV_PARS_DICT)
 
-    # Define the column names that specify a complete and unique row
-    key_names = ("date", "type", "tlmsid", "scs", "step", "source", "vcdu")
+    arch_vals = get_list_for_matching(cmds_arch_recent)
+    recent_vals = get_list_for_matching(cmds_recent)
 
-    recent_vals = [
-        tuple(
-            row[x].decode("ascii") if isinstance(row[x], bytes) else str(row[x])
-            for x in key_names
-        )
-        for row in cmds_arch_recent
-    ]
-    arch_vals = [
-        tuple(
-            row[x].decode("ascii") if isinstance(row[x], bytes) else str(row[x])
-            for x in key_names
-        )
-        for row in cmds_recent
-    ]
-
-    diff = difflib.SequenceMatcher(a=recent_vals, b=arch_vals, autojunk=False)
+    diff = difflib.SequenceMatcher(a=arch_vals, b=recent_vals, autojunk=False)
```

Comment:
Note that the original code had the names reversed (`recent` vs `arch`). `arch_vals` was coming from `cmds_recent` and `recent_vals` was from `cmds_arch_recent`. The names are fixed now with no change in content of `a` and `b`.

Reply:
Great!  I was trying to get why the right/newer answer wasn't "winning" even if there was a bug that meant the code wasn't recognizing the obs statement was different.  I think this fix explains that.